pytools/graph.py

```diff
@@ -420,4 +421,45 @@ def validate_graph(graph: Mapping[T, Collection[T]]) -> None:
 
 # }}}
 
+
+# {{{
+
+def is_connected(graph: Mapping[T, Collection[T]]) -> bool:
+    """
+    Returns whether all nodes in *graph* are connected, ignoring
+    the edge direction.
+
+    :arg graph: A :class:`collections.abc.Mapping` representing a directed
+        graph. The dictionary contains one key representing each node in the
+        graph, and this key maps to a :class:`collections.abc.Collection` of nodes
+        that are connected to the node by outgoing edges.
```

Comment:
By this point, I think we might be repeating this definition over and over. It might be sensible to refer to a central location instead.

Reply:
What do you think of #166?